.github/workflows/main.yml

```
@@ -83,6 +83,7 @@ jobs:
     permissions:
       contents: read  # for actions/checkout to fetch code
       packages: write  # for setup-dotnet to create package
+      code-quality: write  # for actions/upload-code-coverage
```

actionlint (1.7.12)

[error] unknown permission scope "code-quality". all available permission scopes are "actions", "artifact-metadata", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "models", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)